### PR TITLE
WIP: Allow usage of ActionControllers

### PR DESCRIPTION
**Pull request by heiglandreas**
This commit allows to use action controllers that contain a single action that is executed when the __invoke-method is executed.

### CODE REVIEW

**Comment by zghosts** [comment]
did you rerun composer update after the change?
I'm getting: 
"Warning: The lock file is not up to date with the latest changes in composer.json. You may be getting outdated dependencies. Run update to update them."
composer update or composer update --lock should fix this

**Comment by heiglandreas** [comment]
As this PR doesn't include any changes to composer.json or composer.lock: no, I didn't. Seemed not to be relevant...

**Comment by iansltx** [comment]
Yep, that's pre-existing. We need to land #446 first.

**Comment by zghosts** [comment]
whoops this remark was meant for #446

File: tests/routers/ActionControllerRouteTest.php [hunk 1]
```diff
@@ -0,0 +1,173 @@
...
+     * @throws Exception
+     */
+    public function testDispatch(array $config, $controller, Request $request, $expectedException = false, $expectedExceptionCode = false)
+    {
+        $db = new PDO('sqlite:memory:');
+
+        $route = new ActionControllerRoute($controller);
+
+        try {
+            $this->assertEquals('val', $route->dispatch($request, $db, $config));
```
Comment (iansltx) [inline tests/routers/ActionControllerRouteTest.php]: I believe you need to fail here if you expected an exception and didn't get one?

File: tests/routers/ActionControllerRouteTest.php [hunk 2]
```diff
@@ -0,0 +1,173 @@
...
+        $request->expects($this->any())
+                ->method('getUrlElement')
+                ->with(1)
+                ->will($this->returnValue($urlElement));
+
+        return $request;
+    }
+}
+
+class TestController5
```
Comment (iansltx) [inline tests/routers/ActionControllerRouteTest.php]: Should these be properly namespaced/autoloaded rather than shoved into the same file?

**Comment by heiglandreas** [comment]
Thanks for those comments. I was thinking also about injecting the Request into __invoke so thanks for. confirming that. A second thought that crossed my mind just yesterday was whether we shoumd try to implement PSR7/PSR15 here as well?

**Comment by iansltx** [comment]
+1 for 7+15. Personal preference for Slim\Http request and response objects but I can live with Diactoros too.

**Comment by heiglandreas** [comment]
Given that there was a personal intersection between Slim and joind.in I find slim/http a felicitous solution

**Comment by svpernova09** [comment]
@heiglandreas What's the plan regarding @iansltx's review here?

**Comment by svpernova09** [comment]
Closing due to inactivity, feel free to pick this back up at your leisure.

**Pull request closed without merging**